buster/llm_utils/embeddings.py

```diff
@@ -1,32 +1,42 @@
 import logging
 from functools import lru_cache
+from typing import Optional
 
 import numpy as np
 import pandas as pd
 from openai import OpenAI
-from tqdm.contrib.concurrent import process_map
+from tqdm.contrib.concurrent import thread_map
 
 logger = logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
-client = OpenAI()
 
+def get_openai_embedding_constructor(client_kwargs: Optional[dict] = None, model: str = "text-embedding-ada-002"):
+    if client_kwargs is None:
+        client_kwargs = {}
+    client = OpenAI(**client_kwargs)
 
-@lru_cache
-def get_openai_embedding(text: str, model: str = "text-embedding-ada-002") -> np.array:
-    try:
-        text = text.replace("\n", " ")
-        response = client.embeddings.create(
-            input=text,
-            model=model,
-        )
-        embedding = response.data[0].embedding
-        return np.array(embedding, dtype="float32")
-    except Exception as e:
-        # This rarely happens with the API but in the off chance it does, will allow us not to loose the progress.
-        logger.exception(e)
-        logger.warning(f"Embedding failed to compute for {text=}")
-        return None
+    @lru_cache
+    def embedding_fn(text: str, model: str = model) -> np.array:
+        try:
+            text = text.replace("\n", " ")
+            response = client.embeddings.create(
+                input=text,
+                model=model,
+            )
+            embedding = response.data[0].embedding
+            return np.array(embedding, dtype="float32")
+        except Exception as e:
+            # This rarely happens with the API but in the off chance it does, will allow us not to loose the progress.
+            logger.exception(e)
+            logger.warning(f"Embedding failed to compute for {text=}")
+            return None
+
+    return embedding_fn
+
+
+# default embedding function
+get_openai_embedding = get_openai_embedding_constructor()
```

Comment:
I thought the point was to fix this and have a default config with smaller timeout?

Reply:
ah sure, we can set it to a more acceptable default, for me the "deafult" was no deafault and a user could adapt it if preferred